src/main/java/io/vertx/ext/camel/impl/CamelBridgeImpl.java

```diff
@@ -0,0 +1,137 @@
+/*
+ *  Copyright (c) 2011-2015 The original author or authors
+ *  ------------------------------------------------------
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  and Apache License v2.0 which accompanies this distribution.
+ *
+ *       The Eclipse Public License is available at
+ *       http://www.eclipse.org/legal/epl-v10.html
+ *
+ *       The Apache License v2.0 is available at
+ *       http://www.opensource.org/licenses/apache2.0.php
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ */
+package io.vertx.ext.camel.impl;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.eventbus.MessageConsumer;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.vertx.ext.camel.*;
+import org.apache.camel.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * The implementation of the camel bridge.
+ *
+ * @author <a href="http://escoffier.me">Clement Escoffier</a>
+ */
+public class CamelBridgeImpl implements CamelBridge {
+
+  private final CamelContext camel;
+  private final List<Consumer> camelConsumers = new ArrayList<>();
+  private final List<Producer> camelProducers = new ArrayList<>();
+  private final List<MessageConsumer> vertxConsumers = new ArrayList<>();
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CamelBridgeImpl.class);
+
+
+  public CamelBridgeImpl(Vertx vertx, CamelBridgeOptions options) {
+    Objects.requireNonNull(vertx);
+    Objects.requireNonNull(options);
+    this.camel = options.getCamelContext();
+    Objects.requireNonNull(camel);
+
+    for (InboundMapping inbound : options.getInboundMappings()) {
+      // camel -> vert.x
+      createInboundBridge(vertx, inbound);
+    }
+
+    for (OutboundMapping outbound : options.getOutboundMappings()) {
+      // vert.x -> camel
+      createOutboundBridge(vertx, outbound);
+    }
+  }
+
+  private void createOutboundBridge(Vertx vertx, OutboundMapping outbound) {
+    Endpoint endpoint = validate(outbound);
+
+    Producer producer;
+    try {
+      producer = endpoint.createProducer();
+      camelProducers.add(producer);
+    } catch (Exception e) {
+      throw new IllegalStateException("The endpoint " + outbound.getUri() + " does not support producers");
```

Comment:
perhaps we should at least keep the `e.getMessage()` in the IllegalStateException message ?